util/workers/worker.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2023 PingCAP, Inc.
```

Comment:
@hawkingrei write some utilities for the worker pool, maybe you can reuse the code?

Reply:
That's great work, but seems it does not support dynamically run consumer goroutines, but we can't determine the concurrency without a prebuilt task slice.

Reply:
We have a SPMCPool to support the product-consumer goroutine pool. Welcome to communicate with me.